app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -465,7 +466,7 @@ def repository_params
 
       keys += [:label, :content_type] if params[:action] == "create"
       if params[:action] == 'create' || @repository.custom?
-        keys += [:url, :gpg_key_id, :ssl_ca_cert_id, :ssl_client_cert_id, :ssl_client_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name]
+        keys += [:url, :gpg_key_id, :ssl_ca_cert_id, :ssl_client_cert_id, :ssl_client_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name, :description]
```

Comment:
Is there any reason to no allow setting this on Redhat repos provided via a manifest? I'm not sure if anyone would want to but I wouldn't have thought that would be a problem.

Reply:
What would be your suggestion here? Add the :description key to line 467?

Also, as a side note, why do we include the if statement after the keys on line 467 but before the keys on line 468/469? Is there a reason, or should we adjust this for consistency?

Reply:
I would add it to 
https://github.com/Katello/katello/blob/55627f22138092794538db079c5cdf2cd0741709/app/controllers/katello/api/v2/repositories_controller.rb#L461-L464

L467 would only allow modifying it when creating.

> Also, as a side note, why do we include the if statement after the keys on line 467 but before the keys on line 468/469? Is there a reason, or should we adjust this for consistency?

They are different if statements, one is during create only and the other is during create or custom repos. We wouldn't want anyone modifying the URL of a repo provided via a manifest or changing the label after creating.

Reply:
What I mean is, why don't we do:
'''ruby
  if params[:action] == "create"
    keys += [:label, :content_type]
  end
  if params[:action] == 'create' || @repository.custom?
    keys += [:url, :gpg_key_id, :ssl_ca_cert_id, :ssl_client_cert_id, :ssl_client_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name, :description]
  end
'''
Does that make more sense?

Reply:
Forgive me if I've missed another difference, but isn't that just a different syntax? Currently, we have a one-line if statement and your suggestion is a multi-line if statement.

Reply:
No, you didn't miss anything. I only say that because the two if statements are written differently. Since I am new, I am trying to understand what our policy is around consistent code. I'm approaching this from a perspective of readability.

Why would we not just do this for the sake of consistency? Unless I am misunderstanding what these lines of code are doing. :)
```ruby
keys += [:label, :content_type] if params[:action] == "create"
keys += [:url, :gpg_key_id, :ssl_ca_cert_id, :ssl_client_cert_id, :ssl_client_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name] if params[:action] == 'create' || @repository.custom?
```

Reply:
Generally, people seem to write 1 line if statement if it's short and multiline if its longer or more complex.

I do take your point about readability, I have no strong feelings regarding which type it should be, others might I suggest you update it however you see fit and see if anyone objects ;)

Reply:
Yeah, i think its fine to allow this for redhat repos, since its not something that 'comes from redhat'.  If a user wants to add some information for their own purposes, that should be fine.